File: stack/Evaluate Reverse Polish Notation.py
def evalRPN(tokens: list[str]) -> int:
    stack = []

    for i in range(len(tokens)):
        t = tokens[i]
        if t.strip('-').isdigit():
            stack.append(int(t))
        else:
            r = stack.pop()
            l = stack.pop()
            if t == '+':
                stack.append(l + r)
            elif t == '-':
                stack.append(l - r)
            elif t == '/':
                stack.append(int(l / r))
            else:
                stack.append(l * r)

    return stack.pop()

print(evalRPN(["10","6","9","3","+","-11","*","/","*","17","+","5","+"]))
print(evalRPN(["3","-4","+"]))

File: stack/Evaluate Reverse Polish Notation.cs
public class Solution
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> nums = new Stack<int>();

        foreach (string token in tokens)
        {
            if (token == "+")
            {
                int r = nums.Pop();
                int l = nums.Pop();
                int res = l + r;
                nums.Push(res);
            }
            else if (token == "-")
            {
                int r = nums.Pop();
                int l = nums.Pop();
                int res = l - r;
                nums.Push(res);
            }
            else if (token == "/")
            {
                int r = nums.Pop();
                int l = nums.Pop();
                int res = l / r;
                nums.Push(res);
            }
            else if (token == "*")
            {
                int r = nums.Pop();
                int l = nums.Pop();
                int res = l * r;
                nums.Push(res);
            }
            else
                nums.Push(Convert.ToInt32(token));

        }
        return nums.Pop();
    }
}
